### PR TITLE
Prison Loafer

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Implement's Goonstation's prison loafer; credit to them for this.

Goon's prison loafer takes anything and converts it into a prison loaf; the more stuff you shove into it, the denser and denser the loaf gets---if you can get a loaf dense enough, it is enough to hypothetically break walls---realistically, it's not going to happen, unless you dedicate most of your shift to shoving a ton of things into it (and all at once).

***Not mapped in, currently.***

img

Related
- Adds George Melonium (a chemical with random effects, largely drug related, though it can heal or damage the user heavily, as we)
- Enables silicate (does nothing for now)
- Adds in liquid spacetime (gives you teleportitis)

:cl: Fox McCloud
add: adds in disciplinary prison loafer
/:cl:

### CODE REVIEW

**Comment by Spacemanspark** [comment]
what

**Comment by DesolateG** [comment]
I'm confused...

**Comment by Fox-McCloud** [comment]
https://en.wikipedia.org/wiki/Nutraloaf

**Comment by Fruerlund** [comment]
Dont see anything usefull for this besides it being /stupid/ IMO. 

👎

**Comment by SheepFighter** [comment]
I like it, though having a chemical named George Melonium might be a bit too much. How about plain old Melonium?

**Comment by uraniummeltdown** [comment]
I loaf it

**Comment by Rumiluntti** [comment]
Fucking memes.
Seems like even more memed up deep fryer from back when you could fry and random shit.

File: code/modules/recycling/loafer.dm [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+			name = "neutron loaf"
+			desc = "Oh good, the flavor atoms in this prison loaf have collapsed down to a a solid lump of neutrons."
+			icon_state = "ploaf4"
+			force = 66
+			throwforce = 66
+			throw_range = 3
+			reagents.add_reagent("polonium", 25)
+		if(8)
+			name = "quark loaf"
+			desc = "This nutritional loaf is collapsing into subatomic flavor particles. It is unfathmomably heavy."
```
Comment (Crazylemon64) [inline code/modules/recycling/loafer.dm]: `unfathomable`

File: code/modules/recycling/loafer.dm [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+			name = "fusion loaf"
+			desc = "Forget fission, the flavor atoms in this loaf are so densely packed now that they are undergoing atomic fusion. What terrifying new flavor atoms might lurk within?"
+			icon_state = "ploaf3"
+			force = 44
+			throwforce = 44
+			throw_range = 4
+			reagents.add_reagent("radium", 25)
+		if(7)
+			name = "neutron loaf"
+			desc = "Oh good, the flavor atoms in this prison loaf have collapsed down to a a solid lump of neutrons."
```
Comment (Crazylemon64) [inline code/modules/recycling/loafer.dm]: Is this supposed to be "oh good" or "oh god"?

File: code/modules/recycling/loafer.dm [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+			return ..()
+
+		var/obj/item/weapon/reagent_containers/food/snacks/prison_loaf/newLoaf = new /obj/item/weapon/reagent_containers/food/snacks/prison_loaf(src)
+		for(var/atom/movable/newIngredient in H)
+			if(newIngredient.reagents)
+				newIngredient.reagents.trans_to(newLoaf, 1000)
+
+			if(istype(newIngredient, /obj/item/weapon/reagent_containers/food/snacks/prison_loaf))
+				var/obj/item/weapon/reagent_containers/food/snacks/prison_loaf/otherLoaf = newIngredient
+				newLoaf.loaf_factor += otherLoaf.loaf_factor * 1.2
```
Comment (Crazylemon64) [inline code/modules/recycling/loafer.dm]: Won't this let you just set this contraption in a cycle, growing at a rate of 1.2^x each time?

File: code/modules/recycling/loafer.dm [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+	var/loaf_recursion = 1
+	var/super_dense = 0
+	list_reagents = list("gravy" = 10, "beans" = 10, "fake_cheese" = 10, "silicate" = 10, "fungus" = 10, "synthflesh" = 10)
+
+/obj/item/weapon/reagent_containers/food/snacks/prison_loaf/Destroy()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/item/weapon/reagent_containers/food/snacks/prison_loaf/proc/update()
+	var/orderOfLoafitude = max(0, min(round(log(8, loaf_factor)), MAXIMUM_LOAF_STATE_VALUE))
```
Comment (Crazylemon64) [inline code/modules/recycling/loafer.dm]: 8
64
512
4096
32768
262144
2097152
16777216
134217728

File: code/modules/recycling/loafer.dm (hunk 3 above)
Comment (Fox-McCloud) [inline code/modules/recycling/loafer.dm]: Hypothetically, yes. If this is ever added, it's never going to be construcstable and will be placed in such a way it'll be very inconvenient to utilize for such...purposes, however.

File: code/modules/recycling/loafer.dm (hunk 2 above)
Comment (Fox-McCloud) [inline code/modules/recycling/loafer.dm]: I believe "good".

**Comment by SamHPurp** [comment]
Memes memes memes, no thanks.

**Comment by uraniummeltdown** [comment]
@SamHPurp Memes are the lifeblood of this game

**Comment by AndriiYukhymchak** [comment]
what the f*ck is this? Doesn't look like med rp to me

**Comment by tigercat2000** [comment]
> everyone complaining about 'its not med rp'

😆

**Comment by sothangel** [comment]
The loafer looks a bit silly and pointless, why?

Call it Melonium rather than George Melonium; What will Silicate do in the future, and the Liquid Spacetime looks like it could be interesting.

**Comment by fludd12** [comment]
Loafer is for feeding prisoners. Nobody said you *have* to construct mega-dense loaves of raw nutriment.

**Comment by Krausus** [comment]
> if you can get a loaf dense enough, it is enough to hypothetically break walls---realistically, it's not going to happen, unless you dedicate most of your shift to shoving a ton of things into it (and all at once).

With the "loafing a loaf magically makes it 20% denser" feature (which is obviously the intended way to make a high-density loaf) you can go from a density of 5 (effectively, the minimum) to the maximum possible density in 95 cycles. Loafing takes 3.6 seconds, traveling through a loop takes some time, so let's round up to an even 5 seconds; that means, with some pipework, you can go from any tiny item to a weaponized loaf in under 8 minutes. "Most of your shift", indeed.

Other fun facts:
- This deletes living mobs. Any living mob that can be put into disposal pipes. It's not even logged.
- It also, of course, deletes anything else that can go into disposal pipes.
- It's an easy source of space fungus, synthflesh, and some food ingredients.

This seems like a powergamer's wet dream. It also seems to be getting a largely negative response, so we probably won't have to worry about it anyway. ¯\\\_(ツ)_/¯

**Comment by Anasari** [comment]
I think Krasus raised a really valid point.

Although I think if we merged this in and just don't use it outside of a-bus, it might be doable.

**Comment by taukausanake** [comment]
I don't think it matters if it deletes living mobs if it's only going to exist in the brig/prison area. Who's going to go through it? Prisoners? It's their fault one way or another but that brings up other problems about having their job opened if it wasn't before. As for anything else that can go into disposal pipes, isn't that the point?

**Comment by Spacemanspark** [comment]
If this is an admin thing ONLY and nothing else, I suppose.
Otherwise I vehemently disagree with this. I thought this was a joke at first.

**Comment by theColdflame** [comment]
>porting memes

**Comment by theColdflame** [comment]
I don't actually play paradise but is George Melons  even remotely relevant

**Comment by FalseIncarnate** [comment]
This just reminds me of when you could put PDAs and such on plates and render them entirely unrecoverable, except now it includes doing the same but with people.

Not a fan unless it goes through major changes (to the point of practically being a completely different thing).

**Comment by SamHPurp** [comment]
> I thought this was a joke at first.

This describes a lot lately.

**Comment by Birdtalon** [comment]
Seems like we're going through a loaf hate relationship here.

**Comment by davidchan13** [comment]
I like the concept and the loafer on goon but it needs to be toned down and given a bit more serious take. The chef cramming garbage, bad food or other 'mistakes' down the designated disposal pipe (or just having it as an alternative gibber) would be fun and interesting, it being able to kill people without a trace, not so much. Its the recycler on crack that doesn't even require an emag to become lethal.

It not being mapped isn't doing any favors, it requires disposal pipes to work so the Chef can't just drop it where he chooses, he needs assistance from Atmos or Cargo. Good luck with that. Make a map of how/where the loafer loop will run, this isn't something that can just added to the kitchen disposal loop as it's going to break any mail order to and from the kitchen as they'll automatically get loafed and sent to disposals.

:-1: in the current state of things.

File: code/modules/recycling/loafer.dm (hunk 3 above)
Comment (davidchan13) [inline code/modules/recycling/loafer.dm]: Just because the loafer can't be moved doesn't mean the pipes around it couldn't be.

**Comment by Draconic-saint** [comment]
Seems like this is a very poorly adjusted port. curb the density shenanigans &, remove the instantly-deleting mobs.

The idea of feeding perma left-overs and chef mistakes is amazing, but not like this.

**Comment by MrFroztee** [comment]
> It also seems to be getting a largely negative response, so we probably won't have to worry about it anyway.

@Krausus Oh no, do you mean all the scary downboats? There's still plenty more people who have shown approval for the project, I don't see why you should neglect their input.

Also, if deleting objects is such a large concern, I don't see why the machine can't be set to blacklist certain items.

**Comment by SamHPurp** [comment]
Up/down votes don't matter. The maintainers have said as much.

**Comment by Krausus** [comment]
> @Krausus Oh no, do you mean all the scary downboats?

No, I mean the majority of the comments being opposed to it and/or listing specific problems.

> There's still plenty more people who have shown approval for the project, I don't see why you should neglect their input.

You mean the 👍 s, I assume. Problem is, anyone can leave a reaction, regardless of how much (if any) time they have spent actually considering what a PR does and its consequences. Reactions are good for a rough estimate of how people feel about a PR, but useless for deciding whether it should get merged.

**Comment by MrFroztee** [comment]
I don't even see what it remotely "memey" about having the loafer, other than the concept that you can throw anything in and make a loaf out of it. Such a concept, I suppose, can be linked to the one that Rumi mentioned, where you can deep fry and consume anything.

I honestly enjoy features like that, because they invite an element of silliness and liberty that people can take to something like making food, akin to making those giant stacking sandwiches.

Speaking of which, when're we taking those hilarious stacking sandwiches out, since we're such a serious MRP server?

**Comment by Lady-Luck** [comment]
Wow you folks are uptight.

I like this PR.

**Comment by Draconic-saint** [comment]
It's not a meme, it's a cool concept.

The fact that you can reloaf for density is a meme.
The fact that you can loaf people & mobs, instantly deleting them is bad.

**Comment by Lady-Luck** [comment]
I think people here have completely forgotten what a meme actually is.

**Comment by jughu** [comment]
doesnt quark loaf litteraly destroy walls when thrown

**Comment by Taaketa** [comment]
Put it in,screw the nay sayers

**Comment by Fox-McCloud** [comment]
As much as I love this, upon looking at it further, this is going to *require* the biological filter to be viable--additionally, it's going to require sprites be moved, and a lot of testing done.

It's more work than it's worth.

**Comment by Aurorablade** [comment]
NOOOOOOOOO

**Comment by Fox-McCloud** [comment]
The branch for this won't be deleted--if anyone wants to bother getting the biofilter to work, properly, and can map this in decently, the code and sprites are available.

**Comment by Lady-Luck** [comment]
rip

**Comment by mrturkeytoe** [comment]
Can you put people in the loafer though?

**Pull request closed without merging**